examples/poc_noresp_ping.nim

```diff
@@ -83,11 +83,13 @@ proc mixnetSimulation() {.async: (raises: [Exception]).} =
   for index, _ in enumerate(nodes):
     noRespPingProto.add(noresp_ping.NoRespPing.new(rng = rng))
 
-    let protoRes = MixProtocol.new(index, numberOfNodes, nodes[index])
-    if protoRes.isErr:
-      error "Mix protocol initialization failed", err = protoRes.error
+    let proto = MixProtocol.new(index, numberOfNodes, nodes[index]).valueOr:
+      error "Mix protocol initialization failed", err = error
       return
-    mixProto.add(protoRes.get())
+
+    proto.init()
+
+    mixProto.add(proto)
```

Comment:
I don't imediately see the purpose of this flow: new, init, then add it to something else.

Perhaps the relevant context is missing because I'm only seeing the diff, but consider adding comments that clarify the reasoning/pattern/flow for the example user